docs/Users/Tutorials/Testing_Bears.rst

```diff
@@ -0,0 +1,137 @@
+How to use LocalBearTestHelper to test your bears
+=================================================
+
+*coala* has an awesome testing framework so you can write tests for bears
+with ease.
+
+You can use the following to test your bears:
+
+- ``LocalBearTestHelper.check_validity``
+- ``LocalBearTestHelper.check_results``
+- ``verify_local_bears``
+
+Understanding through examples
+------------------------------
+
+Let us understand how to write tests for ``MyBear`` in ``some_dir``.
+``My Bear`` checks if a file has less than or equal to 10 lines.
+
+.. code::
+
+    from coalib.results.Result import Result
+    from coalib.bears.LocalBear import LocalBear
+
+
+    class MyBear(LocalBear):
+
+        def run(file,
+                filename,
+                max_number_of_lines: int=10):
+            """
+            Detects if a file has more than 10 lines
+
+            :param max_number_of_lines    Maximum number of lines to be
+                                          allowed for a file.
+            """
+
+            if(len(file)>10):
+                yield Result(self, "Too many lines")
+
+**EXAMPLE 1** using ``verify_local_bear``
+
+.. code::
+
+    from bears.some_dir.MyBear import MyBear
+    from tests.LocalBearTestHelper import verify_local_bear
+
+    good_file = '1\n2\n3\n4\n'.splitlines()
+    bad_file = '1\n2\n3\n4\n5\n6\n7\n8\n9\n10\n11\n'.splitlines()
+
+    MyBearTest = verify_local_bear(MyBear,
+                                   valid_files=(good_file,),
+                                   invalid_files=(bad_file,))
+
+``good_file`` is a file which your bear considers as non-style-violating
+and a ``bad_file`` is one which has at least one error/warning/info.
+We need to write a ``good_file`` which has less than or equal to 10 lines and a
+``bad_file`` which has more than 10 lines and feed them to
+``verify_local_bear`` as input along with your bear (MyBear in this case) and
+a few additional arguments.
+
+.. note::
+
+    ``good_file`` and ``bad_file`` are sequences just like ``file``. A ``file``
+    is a sequence of an input file.
+
+**EXAMPLE 2** using ``LocalBearTestHelper.check_validity``
+
+.. code::
+
+    from queue import Queue
+    from bears.some_dir.MyBear import MyBear
+
+    from tests.LocalBearTestHelper import LocalBearTestHelper
+    from coalib.settings.Section import Section
+    from coalib.settings.Setting import Setting
+
+
+    class MyBearTest(LocalBearTestHelper):
+
+            self.section = Section('name')
+            self.section.append(Setting('max_number_of_lines', '10'))
+        def setUp(self):
+            self.uut = MyBear(self.section, Queue())
+
+        def test_valid(self):
+            self.check_validity(self.uut, ["import os"])
+
+        def test_invalid(self):
+            self.check_validity(self.uut, bad_file, valid=False)
```

Comment:
ok, so i think it would still be worth mentioning that `bad_file` remains the same as in previous example? maybe down below where you explain everything?